components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -202,17 +204,17 @@ void ClearDuplicatedNodes(ObjectIdToNodes* object_id_nodes,
        it_object_id != object_id_nodes->end(); ++it_object_id) {
     const SortedNodes& nodes = it_object_id->second;
     if (nodes.size() > 1) {
-      // Nodes are sorted from oldest to newest, go to the second by age
-      SortedNodes::iterator it_nodes = nodes.begin();
-      ++it_nodes;
-      for (; it_nodes != nodes.end(); ++it_nodes) {
+      // Re-create all group of nodes which have the same object_id
+      for (SortedNodes::iterator it_nodes = nodes.begin();
+           it_nodes != nodes.end(); ++it_nodes) {
         const bookmarks::BookmarkNode* node = *it_nodes;
         // Copy and delete node
         const auto* parent = node->parent();
         size_t original_index = parent->GetIndexOf(node);
         model->Copy(node, parent, original_index);
         model->Remove(node);
-        brave_sync::AddBraveMetaInfo(parent->children()[original_index].get());
+        brave_sync::AddBraveMetaInfoNoOrder(
```

Comment:
Why do we have to add meta info here? Can't we wait for `commit.cc` to add these meta info all together?

Reply:
@darkdh 

Yes, you are right. Metainfo change caused by my commit call 
 `brave_sync::AddBraveMetaInfoNoOrder` in  is overwritten later by call of: 
```
#2 0x55e9d0463718 brave_sync::AddBraveMetaInfo()
#3 0x55e9d23ec30f sync_bookmarks::CreateSpecificsFromBookmarkNode()
#4 0x55e9d23ee68e sync_bookmarks::BookmarkLocalChangesBuilder::BuildCommitRequests()
#5 0x55e9d23df66d sync_bookmarks::BookmarkModelTypeProcessor::GetLocalChanges()
```
So that call is not required and migration PR becomes smaller.

Tested on my faulty profile - it works. But just in case I want to check with @fmarier this change would work for him before merge this into master.